File: packages/r/R/validation.R
#' Validate an OCR backend name
#'
#' @param backend Backend name string.
#' @return Logical indicating if the backend name is valid.
#' @keywords internal
validate_ocr_backend_name <- function(backend) {
  validate_ocr_backend_name_native(backend)
}

#' Validate a language code
#'
#' @param code Language code string.
#' @return Logical indicating if the code is valid.
#' @keywords internal
validate_language_code <- function(code) {
  validate_language_code_native(code)
}

#' Validate an output format
#'
#' @param format Output format string.
#' @return Logical indicating if the format is valid.
#' @keywords internal
validate_output_format <- function(format) {
  validate_output_format_native(format)
}

File: packages/r/src/rust/src/validation.rs
//! Validation function wrappers

pub fn validate_ocr_backend_impl(_backend: &str) -> extendr_api::Result<bool> {
    // OCR backend validation is handled at registration time
    // For now, accept any non-empty string
    Ok(!_backend.is_empty())
}

pub fn validate_language_code_impl(_code_str: &str) -> extendr_api::Result<bool> {
    // Language code validation is flexible - accept standard codes
    Ok(!_code_str.is_empty())
}

pub fn validate_output_format_impl(_format: &str) -> extendr_api::Result<bool> {
    // Check known output formats
    Ok(matches!(_format, "text" | "plain" | "markdown" | "html" | "json" | "element_based" | "unified" | "djot"))
}
